app/models/qc_reception.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# A QcReception makes an entry in qc_receptions for all the requests
+# received TOL consumer on the qc_reception endpoint.
+# Stores the qc data in qc_results table with the associated qc_reception_id
+class QcReception < ApplicationRecord
+  extend NestedValidation
+  has_many :qc_results, dependent: :restrict_with_error
+  # dependent: :restrict_with_error to stop destroying a record in qc_receptions
```

Comment:
Is this commented out code `dependent: :restrict_with_error` needed?

Reply:
Done